lib/srv/db/cloud/users/helpers.go

```diff
@@ -80,23 +89,34 @@ func (m *lookupMap) setDatabaseUsers(database types.Database, users []User) {
 	database.SetManagedUsers(usernames)
 }
 
+func (m *lookupMap) removeIfURIChanged(database types.Database) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	current, ok := m.byName[database.GetName()]
+	if !ok || current.database.GetURI() == database.GetURI() {
+		return
+	}
+	delete(m.byName, database.GetName())
+}
+
 // removeUnusedDatabases removes unused databases by comparing with provided
 // active databases.
 func (m *lookupMap) removeUnusedDatabases(activeDatabases types.Databases) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
-	for database := range m.byDatabase {
+	for database := range m.byName {
 		if isActive := findDatabase(activeDatabases, database); !isActive {
-			delete(m.byDatabase, database)
+			delete(m.byName, database)
 		}
 	}
 }
 
 // findDatabase finds the database object in provided list of databases.
-func findDatabase(databases types.Databases, database types.Database) bool {
+func findDatabase(databases types.Databases, database string) bool {
 	for i := range databases {
-		if databases[i] == database {
+		if databases[i].GetName() == database {
```

Comment:
Is it worth changing `databases` / `activeDatabases` to a map too? Does it get big enough to justify it?

Reply:
ya, I heard some customers have hundreds of databases so definitely worth it. Updated. And actually less code =D. Thanks for the suggestion!